src/main/java/org/swisspush/redisques/util/DefaultRedisProvider.java

```diff
@@ -62,21 +74,54 @@ private Future<RedisAPI> connectToRedis() {
         int redisMaxPoolSize = config.getMaxPoolSize();
         int redisMaxPoolWaitingSize = config.getMaxPoolWaitSize();
         int redisMaxPipelineWaitingSize = config.getMaxPipelineWaitSize();
+        int redisPoolRecycleTimeoutMs = config.getRedisPoolRecycleTimeoutMs();
 
         Promise<RedisAPI> promise = Promise.promise();
-        RedisOptions redisOptions = new RedisOptions()
-                .setConnectionString(createConnectString())
-                .setMaxPoolSize(redisMaxPoolSize)
-                .setMaxPoolWaiting(redisMaxPoolWaitingSize)
-                .setMaxWaitingHandlers(redisMaxPipelineWaitingSize);
-        redisOptions.setPassword((redisAuth == null ? "" : redisAuth));
-        Redis.createClient(vertx, redisOptions).connect(event -> {
-            if (event.failed()) {
-                promise.fail(event.cause());
-            } else {
-                promise.complete(RedisAPI.api(event.result()));
-            }
-        });
+
+        // make sure to invalidate old connection if present
+        if (redis != null) {
+            redis.close();
+        }
+
+        if (connecting.compareAndSet(false, true)) {
+            redis = Redis.createClient(vertx, new RedisOptions()
+                    .setConnectionString(createConnectString())
+                    .setPassword((redisAuth == null ? "" : redisAuth))
+                    .setMaxPoolSize(redisMaxPoolSize)
+                    .setMaxPoolWaiting(redisMaxPoolWaitingSize)
+                    .setPoolRecycleTimeout(redisPoolRecycleTimeoutMs)
+                    .setMaxWaitingHandlers(redisMaxPipelineWaitingSize));
+
+            redis.connect().onSuccess(conn -> {
+                log.info("Successfully connected to redis");
+                client = conn;
+                client.close();
```

Comment:
For me this creates issues after some 2 minutes

![Uploading image.png…]()

Reply:
![image](https://github.com/swisspost/vertx-redisques/assets/8819088/100d3774-fa0a-4270-b7a5-ca333baab598)

See the README. You need to seth the _redisPoolRecycle_TimeoutMs_ value to _-1_